src/lib/services/api-token-service.test.ts

```diff
@@ -243,4 +248,44 @@ describe('When token is added by another instance', () => {
         expect(found).toBeDefined();
         expect(found?.username).toBe(token.tokenName);
     });
+
+    test('should query the db only once for invalid tokens', async () => {
+        jest.useFakeTimers();
+        const { apiTokenService, apiTokenStore } = setup({
+            experimental: {
+                flags: {
+                    queryMissingTokens: true,
+                },
+            },
+        });
+        const spy = jest.spyOn(apiTokenStore, 'get');
+
+        const invalidToken = 'invalid-token';
+        let found: IApiUser | undefined = undefined;
+        for (let i = 0; i < 10; i++) {
+            try {
+                found = await apiTokenService.getUserForToken(
+                    `${invalidToken}-${i % 2}`,
+                );
+            } catch (e) {
+                // ignore
+            }
+            expect(found).toBeUndefined();
+        }
+        expect(spy).toHaveBeenCalledTimes(2);
```

Comment:
Not super readable at a glance the way it's written, but I guess it's called twice because it's called once for the even token and another for the odd token?

I wonder if it would be better to simplify this, even if it costs DRYness.

Reply:
Sure, I think it makes sense, I was just playing around to make sure it worked properly but I realize it was a bit odd especially because of the try catch but also not very clear to see we have 2 tokens. I'll make a small change on it

Reply:
Addressed here: [f323e68](https://github.com/Unleash/unleash/pull/6771/commits/f323e686dbd12e3060dd7ff0df3ea83c60b75a22)